eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -762,49 +772,51 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
+              val fundingPubKey = keyManager.fundingPublicKey(d.commitments.params.localParams.fundingKeyPath, c.fundingTxIndex)
+              val channelAnn = Announcements.makeChannelAnnouncement(nodeParams.chainHash, localAnnSigs.shortChannelId, nodeParams.nodeId, remoteNodeId, fundingPubKey.publicKey, c.remoteFundingPubKey, localAnnSigs.nodeSignature, remoteAnnSigs.nodeSignature, localAnnSigs.bitcoinSignature, remoteAnnSigs.bitcoinSignature)
               if (!Announcements.checkSigs(channelAnn)) {
                 handleLocalError(InvalidAnnouncementSignatures(d.channelId, remoteAnnSigs), d, Some(remoteAnnSigs))
               } else {
                 // We generate a new channel_update because we can now use the scid of the announced funding transaction.
-                val scidForChannelUpdate = Helpers.scidForChannelUpdate(Some(channelAnn), d.shortIds.localAlias)
-                val channelUpdate = Helpers.makeChannelUpdate(nodeParams, remoteNodeId, scidForChannelUpdate, d.commitments, d.channelUpdate.relayFees)
+                val commitments1 = d.commitments.addAnnouncement(channelAnn)
+                val scidForChannelUpdate = Helpers.scidForChannelUpdate(Some(channelAnn), d.aliases.localAlias)
+                val channelUpdate = Helpers.makeChannelUpdate(nodeParams, remoteNodeId, scidForChannelUpdate, commitments1, d.channelUpdate.relayFees)
+                context.system.eventStream.publish(ShortChannelIdAssigned(self, d.channelId, Some(channelAnn), d.aliases, remoteNodeId))
                 // We use goto() instead of stay() because we want to fire transitions.
-                goto(NORMAL) using d.copy(channelAnnouncement = Some(channelAnn), channelUpdate = channelUpdate) storing()
-              }
-            case Some(_) =>
-              if (!announcementSigsSent.contains(remoteAnnSigs.shortChannelId)) {
-                log.info("re-sending announcement_signatures for scid={}", remoteAnnSigs.shortChannelId)
-                val localAnnSigs = Helpers.makeAnnouncementSignatures(nodeParams, d.commitments.params, d.commitments.latest.remoteFundingPubKey, remoteAnnSigs.shortChannelId)
-                announcementSigsSent += remoteAnnSigs.shortChannelId
-                stay() sending localAnnSigs
-              } else {
-                log.debug("ignoring remote announcement_signatures for scid={}, channel is already announced", remoteAnnSigs.shortChannelId)
-                stay()
+                goto(NORMAL) using d.copy(commitments = commitments1, channelUpdate = channelUpdate) storing()
               }
+            case None =>
+              log.warning("cannot generate announcement_signatures for scid={} with fundingTxIndex={}", remoteAnnSigs.shortChannelId, c.fundingTxIndex)
+              trimAnnouncementSigsStashIfNeeded()
+              announcementSigsStash += (remoteAnnSigs.shortChannelId -> remoteAnnSigs)
+              stay()
```

Comment:
It could also be that `remoteAnnSigs.shortChannelId` is wrong or unexisting. I find it a little weird that the condition to add to `announcementSigsStash` and remove from it are not symmetrical:
- we stash when the related commitment is not found
- we unstash when we receive a `splice_locked` from remote.

Shouldn't we instead:
- stash when the corresponding commit exists but is not locally locked (and drop if unexisting)
- unstash when the corresponding commit gets locally locked?

Reply:
> we cannot distinguish those two cases, so we have to assume it's the first one

Doh, that is the part I was missing. We don't know the local scid until it is confirmed.